### PR TITLE
test: sequential receipt of messages in MConnection 

### PR DESCRIPTION
**Pull request by staheri14**
Accomplishes the first task of #1192

### CODE REVIEW

File: p2p/conn/connection_bench_test.go [hunk 1]
```diff
@@ -566,3 +608,114 @@ func BenchmarkMConnection_Multiple_ChannelID(b *testing.B) {
 		})
 	}
 }
+
+// BenchmarkMConnection_Multiple_ChannelID assesses the max bw/send rate
```
Comment (rootulp) [inline p2p/conn/connection_bench_test.go:612]: [nit] comment and function name should match

```suggestion
// TestMConnection_Message_Order_ChannelID assesses the max bw/send rate
```

File: p2p/conn/connection_bench_test.go [hunk 2]
```diff
@@ -566,3 +608,114 @@ func BenchmarkMConnection_Multiple_ChannelID(b *testing.B) {
...
+	msgSize := 1 * kibibyte
+	sendRate := 50 * kibibyte
+	recRate := 50 * kibibyte
+	clientChDesc := []*ChannelDescriptor{
+		{ID: 0x01, Priority: 1, SendQueueCapacity: 10,
+			RecvMessageCapacity: defaultRecvMessageCapacity,
+			RecvBufferCapacity:  defaultRecvBufferCapacity},
+		{ID: 0x02, Priority: 2, SendQueueCapacity: 1,
+			// channel ID 2's send queue capacity is limited to 1;
+			// to enforce a specific send order.
```
Comment (rootulp) [inline p2p/conn/connection_bench_test.go:636]: [uber nit] I think it's slightly easier to understand comments when they are located immediately above the line they describe

```go
{
	ID: 0x02, 
	Priority: 2, 
	// channel ID 2's send queue capacity is limited to 1;
	// to enforce a specific send order.
	SendQueueCapacity: 1, 
```

or on the same line

```go
{
	ID: 0x02, 
	Priority: 2, 
	SendQueueCapacity: 1, // limited to 1 to enforce a specific send order
```

**Review by rootulp — APPROVED** [review]
LGTM

File: p2p/conn/connection_bench_test.go (hunk 2 above)
Comment (staheri14) [reply p2p/conn/connection_bench_test.go]: Sure, resolved https://github.com/celestiaorg/celestia-core/pull/1193/commits/a4b979aaaf31cd6f3b93eb081075e195a630ac2f

File: p2p/conn/connection_bench_test.go (hunk 1 above)
Comment (staheri14) [reply p2p/conn/connection_bench_test.go]: thanks, fixed https://github.com/celestiaorg/celestia-core/pull/1193/commits/a4b979aaaf31cd6f3b93eb081075e195a630ac2f

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**